lib/actions/ssh-direct.ts

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2018 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as sdk from 'balena-sdk';
+import * as Bluebird from 'bluebird';
+import * as globalTunnel from 'global-tunnel-ng';
```

Comment:
Related new finding: proxy is currently not working with Node.js versions >= 10.16.0 (so not just Node 11, 12). Nothing specific to this PR -- the master branch is affected. Ive created this issue: https://github.com/balena-io/balena-cli/issues/1579

Reply:
I have created PR #1582 that uses `global-agent` for Node.js >= 10.16.0, and `global-tunnel-ng` for older versions, addressing issues #1579, #1580 and #1335.

Reply:
PR #1582 was merged, but something that remains unexplained is that I couldn't get `balena ssh` to work behind a proxy (with balenaCloud, not openBalena) when `proxytunnel` is involved. And that was the case both before and after the PR was merged, testing on both Linux and macOS. Therefore, if it would be great if you could test that `balena ssh` works behind a proxy with balenaCloud, even if just on your own machine (Linux?), to check whether it was just me (maybe I was setting something up incorrectly).

Reply:
> I couldn't get `balena ssh` to work behind a proxy (with balenaCloud, not openBalena) with `proxytunnel`

Now I managed to get it working, including basic authentication (username and password). What was happening is that my proxy server was not correctly configured to allow proxying to port 22 + SSL. PR #1685 adds a note to the README file about proxy configuration for ssh access, and makes some changes to address issue #1681.